### PR TITLE
Update assets documentation

### PR DESCRIPTION
**Pull request by atangOSIsoft**
added updates for:
- Additional behavior and rules updates for metadata and references name and id
- Removal of Name from Status

### CODE REVIEW

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 1]
```diff
@@ -126,6 +126,16 @@ Create a new asset with a specified `Id`.
 
 If the asset you are trying to create references an asset type (through the AssetTypeId property) and if the corresponding asset type has a metadata value with the same `Id`, then the name and SDS type code of the metadata value on the asset must be null. If the asset type does not have metadata value with a corresponding `Id`, the name and SDS type code on the asset cannot be null.
 
+To support flexibility, on creation and update, the following rules and behaviors are executed for metadata and stream references on a given asset when that asset is from an asset type.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:129]:
```suggestion
To support flexibility on creation and update, the following rules and behaviors are executed for metadata and stream references on a given asset when that asset is created from an asset type.
```

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 2]
```diff
@@ -126,6 +126,16 @@ Create a new asset with a specified `Id`.
...
+
+| Example                                             | Metadata value on asset type                          |                                              |
+| ----------------------------------------------------| ----------------------------------------------------- | ------------------------------------------------- | 
+| Id = "Metadata Id"​ and Name = Not specified         | Id matches Asset Type Metadata Id | Uses matching Asset Type Metadata Name |
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:133]:
```suggestion
| Id = "Metadata Id"​ and Name = Not specified         | Id matches asset type metadata Id | Uses matching asset type metadata name |
```

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 3]
```diff
@@ -126,6 +126,16 @@ Create a new asset with a specified `Id`.
...
+|                                                     | Id does not match any Asset Type Metadata Id | Id is used as Name |
+| Id = Not specified​ and Name = "Metadata Name"       | Name matches Asset Type Metadata Name | Use matching Asset Type Metadata Id |
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:135]:
```suggestion
| Id = Not specified​ and Name = "Metadata Name"       | Name matches asset type metadata name | Use matching asset type metadata Id |
```
Anthony,  Why do you have "Metadata Name" in quotes? Is this to indicate that this is a parameter? Is Name a parameter?

Comment (atangOSIsoft) [reply Content_Portal/Documentation/Assets/assets-api.md:135]: I think perhaps that was an initial mistake. Thanks for updating.

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 4]
```diff
@@ -126,6 +126,16 @@ Create a new asset with a specified `Id`.
 
 If the asset you are trying to create references an asset type (through the AssetTypeId property) and if the corresponding asset type has a metadata value with the same `Id`, then the name and SDS type code of the metadata value on the asset must be null. If the asset type does not have metadata value with a corresponding `Id`, the name and SDS type code on the asset cannot be null.
 
+To support flexibility on creation and update, the following rules and behaviors are executed for metadata and stream references on a given asset when that asset is created from an asset type.
+
+| Example                                             | Metadata value on asset type                          |                                              |
+| ----------------------------------------------------| ----------------------------------------------------- | ------------------------------------------------- | 
+| Id = "Metadata Id"​ and Name = Not specified         | Id matches Asset Type Metadata Id | Uses matching Asset Type Metadata Name |
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:133]:
```suggestion
| `Id` = ***Metadata Id***​ and Name = Not specified         | `Id` matches the asset type's Metadata Id | Uses matching asset type Metadata Name |
```

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 5]
```diff
@@ -126,6 +126,16 @@ Create a new asset with a specified `Id`.
 
 If the asset you are trying to create references an asset type (through the AssetTypeId property) and if the corresponding asset type has a metadata value with the same `Id`, then the name and SDS type code of the metadata value on the asset must be null. If the asset type does not have metadata value with a corresponding `Id`, the name and SDS type code on the asset cannot be null.
 
+To support flexibility on creation and update, the following rules and behaviors are executed for metadata and stream references on a given asset when that asset is created from an asset type.
+
+| Example                                             | Metadata value on asset type                          |                                              |
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:131]:
```suggestion
| Asset values                                             | Metadata values on the asset type                          |                                              |
```

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 6]
```diff
@@ -126,6 +126,16 @@ Create a new asset with a specified `Id`.
 
 If the asset you are trying to create references an asset type (through the AssetTypeId property) and if the corresponding asset type has a metadata value with the same `Id`, then the name and SDS type code of the metadata value on the asset must be null. If the asset type does not have metadata value with a corresponding `Id`, the name and SDS type code on the asset cannot be null.
 
+To support flexibility on creation and update, the following rules and behaviors are executed for metadata and stream references on a given asset when that asset is created from an asset type.
+
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:130]: ### Examples

File: Content_Portal/Documentation/Assets/assets-api.md (hunk 4 above)
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:133]:
```suggestion
| Id = "Metadata Id"​ and Name = Not specified         | Id matches the asset type's Metadata Id | Uses the matching asset type's Metadata Name |
```

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 7]
```diff
@@ -126,6 +126,16 @@ Create a new asset with a specified `Id`.
...
+| Example                                             | Metadata value on asset type                          |                                              |
+| ----------------------------------------------------| ----------------------------------------------------- | ------------------------------------------------- | 
+| Id = "Metadata Id"​ and Name = Not specified         | Id matches Asset Type Metadata Id | Uses matching Asset Type Metadata Name |
+|                                                     | Id does not match any Asset Type Metadata Id | Id is used as Name |
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:134]:
```suggestion
|                                                     | Id does not match any asset type Metadata Id | Id is used as Name |
```

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 8]
```diff
@@ -126,6 +126,16 @@ Create a new asset with a specified `Id`.
...
+| Id = Not specified​ and Name = "Metadata Name"       | Name matches Asset Type Metadata Name | Use matching Asset Type Metadata Id |
+|                                                     | Name does not match any Asset Type Metadata Name | GUID is generated for Id |
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:136]:
```suggestion
|                                                     | Name does not match any asset type Metadata Name | GUID is generated for Id |
```

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 9]
```diff
@@ -126,6 +126,16 @@ Create a new asset with a specified `Id`.
...
+| Id = "Metadata Id"​ and Name = "Metadata Name"       | Only Id or Name (not both) matches Asset Type Metadata Id or Name | Error. |
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:137]:
```suggestion
| Id = "Metadata Id"​ and Name = "Metadata Name"       | Either Id or Name, but not both, match asset type Metadata Id or Name | Error. |
```

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 10]
```diff
@@ -224,6 +234,19 @@ The response includes a status code and a body.
 
 Create or update an asset with a specified `Id`. If the asset already exists, you can specify an If-Match property in the HTTP request header to ensure that the asset is modified only if its version matches.
 
+To support flexibility, on creation and update, the following rules and behaviors are executed for metadata and stream references on a given asset when that asset is from an asset type.
+
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:238]: ###Examples

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 11]
```diff
@@ -224,6 +234,19 @@ The response includes a status code and a body.
...
+| Example                                             | Metadata value on asset type                          |                                              |
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:239]:
```suggestion
| Asset values                                            | Metadata values on the asset type                          |                                              |
```

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 12]
```diff
@@ -224,6 +234,19 @@ The response includes a status code and a body.
...
+| ----------------------------------------------------| ----------------------------------------------------- | ------------------------------------------------- | 
+| Id = "Metadata Id"​ and Name = Not specified         | Id matches Asset Type Metadata Id | Uses matching Asset Type Metadata Name |
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:241]:
```suggestion
| Id = "Metadata Id"​ and Name = Not specified         | Id matches the asset type's Metadata Id | Uses the matching asset type Metadata Name |
```

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 13]
```diff
@@ -224,6 +234,19 @@ The response includes a status code and a body.
...
+|                                                     | Id matches existing Asset Type Metadata Id | Uses existing matching Metadata Name|
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:242]:
```suggestion
|                                                     | Id matches the existing asset type's Metadata Id | Uses the existing matching Metadata Name|
```

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 14]
```diff
@@ -224,6 +234,19 @@ The response includes a status code and a body.
...
+|                                                     | Id does not match any Asset Type Metadata Id | Id is used as Name |
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:243]:
```suggestion
|                                                     | Id does not match any asset type Metadata Id | Id is used as Name |
```

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 15]
```diff
@@ -224,6 +234,19 @@ The response includes a status code and a body.
...
+| Id = Not specified​ and Name = "Metadata Name"       | Name matches Asset Type Metadata Name | Use matching Asset Type Metadata Id |
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:244]:
```suggestion
| Id = Not specified​ and Name = "Metadata Name"       | Name matches the asset type's Metadata Name | Uses the matching asset type Metadata Id |
```

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 16]
```diff
@@ -224,6 +234,19 @@ The response includes a status code and a body.
...
+|                                                     | Name matches existing Asset Type Metadata Name | Use matching existing Asset Type Metadata Id |
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:245]:
```suggestion
|                                                     | Name matches the existing asset type's Metadata Name | Uses the matching existing asset type Metadata Id |
```

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 17]
```diff
@@ -224,6 +234,19 @@ The response includes a status code and a body.
...
+|                                                     | Name does not match any Asset Type Metadata Name | GUID is generated for Id |
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:246]:
```suggestion
|                                                     | Name does not match any asset type Metadata Name | GUID is generated for Id |
```

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 18]
```diff
@@ -224,6 +234,19 @@ The response includes a status code and a body.
...
+| Id = "Metadata Id"​ and Name = "Metadata Name"       | Only Id or Name (not both) matches Asset Type Metadata Id or Name | Error. |
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:247]:
```suggestion
| Id = "Metadata Id"​ and Name = "Metadata Name"       | Only Id or Name (not both) matches asset type Metadata Id or Name | Error. |
```

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 19]
```diff
@@ -224,6 +234,19 @@ The response includes a status code and a body.
...
+|                                                     | Only Id or Name (not both) matches existing Asset Type Metadata Id or Name | The non-matching Id or Name of the existing Metadata is updated |
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:248]:
```suggestion
|                                                     | Only Id or Name (not both) matches existing asset type Metadata Id or Name | The non-matching Id or Name of the existing Metadata is updated |
```

**Review by lasato-OSI — CHANGES_REQUESTED** [review]
Hi Anthony,
Here are the changes I made:
- I changed _Asset Type_ to _asset type._
- I guess I'm still unclear when it says: Id = "Metadata Id" if that means that the actual value of Id is _Metadata Id_ or if the values of these two parameters are identical. 
-  When it says something like: _Name matches existing asset type Metadata Id_, is _Metadata Id_ referring to specific "parameters"? (sorry, I may not be using the correct term here.) I'm assuming it does, so I left it as _Metadata Id_, not _metadata_ _Id_.
- I changed the heading of the first column from _Examples_ to _Asset values_ which I think is what this column represents. We also need a heading for the third column, something like: How references are resolved? I added a heading, _Examples_, before the table.

I may need to go through the file again and clean it up for consistency.

File: Content_Portal/Documentation/Assets/assets-api.md (hunk 5 above)
Comment (atangOSIsoft) [reply Content_Portal/Documentation/Assets/assets-api.md:131]: That change looks fine to me.

File: Content_Portal/Documentation/Assets/assets-api.md (hunk 15 above)
Comment (atangOSIsoft) [reply Content_Portal/Documentation/Assets/assets-api.md:244]: Looks good. Thanks

File: Content_Portal/Documentation/Assets/assets-api.md (hunk 16 above)
Comment (atangOSIsoft) [reply Content_Portal/Documentation/Assets/assets-api.md:245]: Looks good. Thanks

**Pull request merged**